### ISSUE TITLE
Support IP_ADD_MEMBERSHIP on lx

### ISSUE THREAD

**Issue by rzezeski** [joyent/illumos-joyent#93]
I recently received a request to support the `IP_ADD_MEMBERSHIP` socket option on lx. I'm curious if this feature might be useful to others and how much effort it would be to implement. It looks like this would be helpful towards fixing #90.

```
# ./socat - UDP4-DATAGRAM:224.255.0.1:6666,bind=:6666,ip-add-membership=224.255.0.1:192.168.241.82
2016/01/18 15:48:12 socat[32072.140737471710976] E setsockopt(5, 0, 35, {0x0100ffe0,0x52f1a8c0,0}, 12): Invalid argument
```

```
# strace -e trace=setsockopt ./socat - UDP4-DATAGRAM:224.255.0.1:6666,bind=:6666,ip-add-membership=224.255.0.1:192.168.241.82
setsockopt(5, SOL_IP, IP_ADD_MEMBERSHIP, "\340\377\0\1\300\250\361R\0\0\0\0", 12) = -1 EINVAL (Invalid argument)
2016/01/18 16:01:46 socat[32177.140737471710976] E setsockopt(5, 0, 35, {0x0100ffe0,0x52f1a8c0,0}, 12): Invalid argument
+++ exited with 1 +++
```

**Comment by pfmooney** [comment]
Filed as OS-5096

**Comment by pfmooney** [comment]
Fixed in joyent/illumos-joyent@713d20cf0dd269ec8a2fa1e38c976b79d88373cd

Thanks
